dubbo-metadata/dubbo-metadata-api/src/main/java/org/apache/dubbo/metadata/definition/ServiceDefinitionBuilder.java

```diff
@@ -61,13 +64,18 @@ public static FullServiceDefinition buildFullDefinition(final Class<?> interface
     public static <T extends ServiceDefinition> void build(T sd, final Class<?> interfaceClass) {
         sd.setCanonicalName(interfaceClass.getCanonicalName());
         sd.setCodeSource(ClassUtils.getCodeSource(interfaceClass));
+        Annotation[] classAnnotations = interfaceClass.getAnnotations();
+        sd.setAnnotations(classAnnotations == null ? new ArrayList<>() : Arrays.asList(classAnnotations));
```

Comment:
Code clean. empty List can use `java.util.Collections#emptyList`. 
Other place also follow this.